symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/OboExampleMain.java

```diff
@@ -0,0 +1,28 @@
+package com.symphony.bdk.examples;
+
+import com.symphony.bdk.core.SymphonyBdk;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.exception.AuthInitializationException;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.config.exception.BdkConfigException;
+
+import com.symphony.bdk.gen.api.model.V4Message;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class OboExampleMain {
+
+  private static final String STREAM = "2IFEMquh3pOHAxcgLF8jU3___ozwgwIVdA";
+  private static final String MESSAGE = "<messageML>Hello, World!</messageML>";
+
+  public static void main(String[] args) throws BdkConfigException, AuthInitializationException, AuthUnauthorizedException {
+
+    final SymphonyBdk bdk = SymphonyBdk.fromSymphonyDir("config.yaml");
+
+    final AuthSession oboSession = bdk.obo("thibault.pensec");
+
+    final V4Message message = bdk.messages().obo(oboSession).send(STREAM, MESSAGE);
```

Comment:
What about bdk.obo(oboSession).messages()

That you first tell who you are and then do something

It would be more consistent to have bdk.obo().messages() then bdk.obo().streams()...
This way you can discover the services available via OBO using autocomplete

But then we already have a method obo that is used to create the auth session, as a user of the BDK should I care about the auth session or should the BDK handle it for me?

it could be `bdk.obo("thibault.pensec").messages().send(...)`
and bdk.messages() is actually a shortcut for bdk.bot().messages()

You might also wonder if the BDK should authenticate when created or only on first use

Looking at the auth code, the AuthSession is not thread safe (but I'm not sure it is a problem we might just ending up doing multiple refresh at the same time but if we end up with valid tokens in the end it could be fine)